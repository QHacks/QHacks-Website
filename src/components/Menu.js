import React from "react";
import ScrollSpy from "react-scrollspy";
import colored from "../assets/img/colored.svg";

const Menu = (props) => {
  return (
    <div
      css={{
        display: "grid",
        gridTemplateColumns: "min-content auto"
      }}
    >
      <img
        css={{
          paddingTop: "20px",
          paddingBottom: "20px",
          transition: "0.5s",
          "@media(min-width:820px)": {
            opacity: props.imgCss ? "1" : "0"
          }
        }}
        src={colored}
      />
      <div
        align="right"
        css={{
          paddingTop: "32px",
          paddingBottom: "32px",
          height: "84px",
          display: "grid",
          transition: "1s",
          marginLeft: "auto",
          "@media(max-width: 820px)": {
            display: "none"
          },
          " li": {
            listStyleType: "none"
          }
        }}
      >
        <ScrollSpy
          css={{
            display: "grid",
            gridColumnGap: "50px",
            gridTemplateColumns: props.menuItems
              .map(() => "min-content ")
              .reduce((a, b) => a + b)
          }}
          items={props.menuItems}
          currentClassName="is-current"
        >
          {props.menuItems.map((label) => (
            <li
              key={label}
              css={{
                ":hover": {
                  " a": {
                    color: "#00205b"
                  },
                  ">div": {
                    width: "100%",
                    height: "2px"
                  }
                }
              }}
            >
              <a
                css={{
                  lineHeight: "1.43",
                  textAlign: "center",
                  paddingLeft: "2px",
                  paddingRight: "2px",
                  textTransform: "uppercase"
                }}
                href={`#${label}`}
              >
                {label}
              </a>
              <div
                css={{
                  width: "100%",
                  height: "2px",
                  transition: "0.5s",
                  marginLeft: "auto",
                  marginRight: "auto",
                  backgroundColor: "#00205b",
                  width: "0px"
                }}
              />
            </li>
          ))}
        </ScrollSpy>
      </div>
    </div>
  );
};

export default Menu;
