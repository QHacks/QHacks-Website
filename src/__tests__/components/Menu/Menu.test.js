import React from "react";
import Menu from "../../../components/Menu";
import { shallow } from "enzyme";

describe("Menu", () => {
  it("renders correctly", () => {
    expect(
      shallow(<Menu breadcrumbs={["THIS", "IS", "A", "TEST"]} />)
    ).toMatchSnapshot();
  });
});
