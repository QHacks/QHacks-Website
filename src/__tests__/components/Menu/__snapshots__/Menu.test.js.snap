// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div
  css={
    Object {
      "display": "grid",
      "gridTemplateColumns": "min-content auto",
    }
  }
>
  <img
    css={
      Object {
        "@media(min-width:820px)": Object {
          "opacity": "0",
        },
        "paddingBottom": "20px",
        "paddingTop": "20px",
        "transition": "0.5s",
      }
    }
    src="test-file-stub"
  />
  <div
    align="right"
    css={
      Object {
        " li": Object {
          "listStyleType": "none",
        },
        "@media(max-width: 820px)": Object {
          "display": "none",
        },
        "display": "grid",
        "height": "84px",
        "marginLeft": "auto",
        "paddingBottom": "32px",
        "paddingTop": "32px",
        "transition": "1s",
      }
    }
  >
    <Scrollspy
      componentTag="ul"
      css={
        Object {
          "display": "grid",
          "gridColumnGap": "50px",
          "gridTemplateColumns": "min-content min-content min-content min-content ",
        }
      }
      currentClassName="is-current"
      items={
        Array [
          "THIS",
          "IS",
          "A",
          "TEST",
        ]
      }
      offset={0}
      onUpdate={[Function]}
      style={Object {}}
    >
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="THIS"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="#THIS"
        >
          THIS
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="IS"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="#IS"
        >
          IS
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="A"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="#A"
        >
          A
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
      <li
        css={
          Object {
            ":hover": Object {
              " a": Object {
                "color": "#00205b",
              },
              ">div": Object {
                "height": "2px",
                "width": "100%",
              },
            },
          }
        }
        key="TEST"
      >
        <a
          css={
            Object {
              "lineHeight": "1.43",
              "paddingLeft": "2px",
              "paddingRight": "2px",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
          href="#TEST"
        >
          TEST
        </a>
        <div
          css={
            Object {
              "backgroundColor": "#00205b",
              "height": "2px",
              "marginLeft": "auto",
              "marginRight": "auto",
              "transition": "0.5s",
              "width": "0px",
            }
          }
        />
      </li>
    </Scrollspy>
  </div>
</div>
`;
